lib/web/purchases_flutter_web.dart

```diff
@@ -314,6 +317,21 @@ class PurchasesFlutterPlugin {
     _callInstanceMethod('setDisplayName', [displayName]);
   }
 
+  Future<Map<String, dynamic>> _getVirtualCurrencies() async =>
+    await _getMapFromInstanceMethod('getVirtualCurrencies', []);
+
+  Future<void> _invalidateVirtualCurrenciesCache() async {
+    _getInstance().callMethod('invalidateVirtualCurrenciesCache'.toJS);
+  }
+
+  Future<Map<String, dynamic>?> _getCachedVirtualCurrencies() async {
+    final cachedVirtualCurrencies = _getInstance().callMethod('getCachedVirtualCurrencies'.toJS);
+    if (cachedVirtualCurrencies == null) {
+      return null;
+    }
+    return _convertJsRecordToMap(cachedVirtualCurrencies);
```

Comment:
Hmm can we use `_getNullableMapFromInstanceMethod` here?

Reply:
Ohh right! Hmm I think not TBH... most of our functions are asynchronous, so it makes sense for them to be promises. So yeah, I think this is good then, thanks for explaining!

Reply:
@tonidero I've extracted that logic in 5dbcda35dfd429fa01b0339a2157898bcead8dea, which will let us re-use this in any future sync function calls that return nullable maps 🙂